From 56928adb02a4b5c14a058f2645052641484c12b0 Mon Sep 17 00:00:00 2001
From: Wan-Teh Chang <wtc@google.com>
Date: Thu, 7 Apr 2022 20:47:37 -0700
Subject: [PATCH] Don't use a non-constant expr as union initializer

Don't use a non-constant expression as a union initializer. Fix the
Visual Studio 2017 compiler warning:

..\src\reformat.c(1103): warning C4204:
nonstandard extension used: non-constant aggregate initializer

Fix https://github.com/AOMediaCodec/libavif/issues/914.
---
 src/reformat.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/reformat.c b/src/reformat.c
index 1e3d3653..ea48196a 100644
--- a/src/reformat.c
+++ b/src/reformat.c
@@ -1096,11 +1096,12 @@ static avifResult avifRGBImageToF16(avifRGBImage * rgb)
     for (uint32_t j = 0; j < rgb->height; ++j) {
         uint16_t * pixel = pixelRowBase;
         for (uint32_t i = 0; i < rgb->width * channelCount; ++i, ++pixel) {
-            const union
+            union
             {
                 float f;
                 uint32_t u32;
-            } f16 = { .f = *pixel * multiplier };
+            } f16;
+            f16.f = *pixel * multiplier;
             *pixel = (uint16_t)(f16.u32 >> 13);
         }
         pixelRowBase += stride;
